.travis.yml

```diff
@@ -18,38 +7,66 @@ jobs:
       osx_image: xcode11.2
       env:
         - MACOSX_DEPLOYMENT_TARGET=10.14
-        - PATH="${HOME}/Library/Python/2.7/bin:${PATH}"
         - MATRIX_EVAL=""
+      install:  # Overrides install step below
+        # pyenv doesn't exist on this CI image, so use virtualenv to set Python version
+        - pip3 install virtualenv
+        - virtualenv -p python3 ~/venv
+        - source ~/venv/bin/activate
+
     - os: linux
       dist: bionic
       compiler: gcc
       env:
-        - GCC_VER="9"
+        - GCC_VER="10"
         - MATRIX_EVAL="CC=gcc-${GCC_VER} && CXX=g++-${GCC_VER}"
 
       addons:
         apt:
           sources:
             - ubuntu-toolchain-r-test
           packages:
-            # I couldn't get ${GCC_VER} in here successfully
-            - gcc-9
-            - g++-9
+            - gcc-10
+            - g++-10
             - doxygen
-            - python3
-            - python3.8
       after_script:
         - bash <(curl -s https://codecov.io/bash) -x /usr/bin/gcov-${GCC_VER}
     - os: linux
       dist: bionic
       compiler: clang
+      before_install:  # Overrides before_install step below
+        # Add LLVM repo and GPG key so we can install clang-10
+        - wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -
+        - echo "deb http://apt.llvm.org/bionic/ llvm-toolchain-bionic-10 main" | sudo tee -a /etc/apt/sources.list
+        - sudo apt-get -qq update
+        - sudo apt-get -y install clang-10
       env:
-        - MATRIX_EVAL="CC=clang-8 && CXX=clang++-8"
-      addons: { apt: { packages: ['doxygen', 'clang-8'] } }
+        - CLANG_VER="10"
+        - MATRIX_EVAL="CC=clang-${CLANG_VER} && CXX=clang++-${CLANG_VER}"
+      addons: { apt: { packages: ['doxygen'] } }
 
+before_install:
+  # Empty. We are using this step to add sources (as in addons/apt/sources) 
+  # that Travis disallows, for example, the LLVM repo.
+
+install:
+  # Set default version of Python to 3: some Conan packages require this.
```

Comment:
I think that the idea that some Conan packages require Python 3 is now untrue. 

**nasm/2.14**, the package that was causing the previous issue, had some comments written in Unicode, and Python 2 could not read the file. That package has been fixed, and Python 2 can now read it.

The conan-io maintainers have added a hook (KB-H047) to their PR process that will prevent this problem in the future.

While it's possible that the conan-io maintainers could break Python 2 compatibility in the future, right now I think it's unnecessary to keep any of this code that sets Python to version 3, and without it we get clearer code.

Reply:
I just received word from one of the Conan-io maintainers that they plan to remove Python 3 support entirely for Conan 2.0. For now, I think we should retain all of this code that sets the default interpreter to 3.